tests/bluetooth/mesh_shell/src/main.c

```diff
@@ -11,12 +11,11 @@
 #include <misc/printk.h>
 #include <zephyr.h>
 
-#include <shell/legacy_shell.h>
+#include <shell/shell.h>
+#include <shell/shell_uart.h>
```

Comment:
This is not needed anymore. 
Backend is selected by Kconfig and its initialization is realized in `shell_uart.c` you do not need to do it in main.c file.